src/Compilers/Core/Portable/Emit/EmitOptions.cs

```diff
@@ -248,6 +292,15 @@ internal void ValidateOptions(DiagnosticBag diagnostics, CommonMessageProvider m
             {
                 diagnostics.Add(messageProvider.CreateDiagnostic(messageProvider.ERR_InvalidSubsystemVersion, Location.None, SubsystemVersion.ToString()));
             }
+
+            try
+            {
+                IncrementalHash.CreateHash(HashAlgorithm).Dispose();
+            }
+            catch
```

Comment:
>catch [](start = 12, length = 5)

A catch-all like this is a bit scary.

Reply:
We are calling to external library (BCL). Any exception that happens means the hash algorithm is bad or there is a bug in BCL. Either way we report a diagnostic. Sure, we can hit OOM, or stack overflow, but such exceptions will be rethrown anyways even if we catch them here.

Reply:
Maybe add this explanation as a comment

---
In reply to: [167302067](https://github.com/dotnet/roslyn/pull/24711#discussion_r167302067) [](ancestors = 167302067)